src/common/history.c

```diff
@@ -872,9 +872,15 @@ int dt_history_copy_and_paste_on_selection(int32_t imgid, gboolean merge, GList
 
 void dt_history_compress_on_image(int32_t imgid)
 {
-  // make sure the right history is in there:
+  // ok, just make sure the current history is written!
   dt_dev_write_history(darktable.develop);
 
+  // We use the global darktable.develop so when we want to use a specific image
```

Comment:
But then we have:

```
void dt_history_compress_on_image_and_reload(int32_t imgid)
{
  dt_history_compress_on_image(imgid);

  /* if current image in develop reload history */
  if(dt_dev_is_current_image(darktable.develop, imgid))
```
The condition above is always TRUE now. Worth checking that this is not broken now? Or remove the conditional?

Reply:
Right. Probably we can remove `dt_history_compress_on_image_and_reload` and directly use  `dt_history_compress_on_image`.

Reply:
@jenshannoschwalm  : Any news on that?

Reply:
I am still hunting for a history leak bug not yet understood.

Reply:
The second commit
1. Moves more stuff from lib/history.c to common/history.c for refactoring
2. the compress_and_reload also keeps the selected position in the history stack
3. The lighttable compress history button has a check box -- also as a feedback.
4. I couldn't produce history leaks any more.
5. For me the compressed history always is ok.

**BUT** there is one remaining problem with the lighttables compression.
Try this:
Select some pictures in the lighttable and do the compression. Now leave darktable and start it again. The compressed histories are ok as expected.

If you do the compression and open **any** picture right after that you will see the history is **wrong**, some entries have found their way into the history.

Could you give me a hint? Do we need to 'invalidate' the darktable.develop or its history somehow?